src/order.py

```diff
@@ -1,169 +1,52 @@
-          "kind": "''' + str(self["kind"]) + '''",
-          "partiallyFillable": ''' + str(self["partiallyFillable"]).lower() + ''',
-          "signature": "''' + str(signed_message.signature.hex()) + '''",
-          "signingScheme": "ethsign",
-          "sellTokenBalance": "''' + str(self["sellTokenBalance"]) + '''",
-          "buyTokenBalance": "''' + str(self["buyTokenBalance"]) + '''",
-          "from": "''' + public_key + '''"
-        }'''
-        return json.loads(json_str)
-
-    def place_order(self):
-        self.fee_amount = self.get_fee()
-        order_bytes = self.signable_bytes(domain)
-        order_hash = w3.keccak(order_bytes)
-        message = encode_defunct(primitive=order_hash)
-        signed_message = w3.eth.account.sign_message(message, private_key=private_key)
-        # Build Json out of Order + signed_message.signature.hex() + public_address
-        order_json = self.to_json(signed_message, public_address)
-        r = requests.post(base_url + "orders", json=order_json)
-        # TODO - log status code.
-
-    def counter_order(self):
-        # TODO - stop accessing fields by string keys.
-        res = Order(
-            sell_token=self.buy_token,
-            buy_token=self.sell_token,
-            receiver=public_address,
-            sell_amount=self.buy_amount,
-            buy_amount=self.sell_amount,
-            # 2 minutes in the future.
-            valid_to=int(time.time()) + 120,
-            kind=self.opposite_kind(),
+import domains
+from JSONEncoder import BytesJSONEncoder
+
+
+class Order(EIP712Message):
+    _name_: "string" = domains.name
+    _version_: "string" = domains.version
+    _chainId_: "uint256" = domains.chain_id
+    _verifyingContract_: "address" = domains.verifying_contract
+
+    sellToken: "address"
+    buyToken: "address"
+    receiver: "address" = domains.public_address
+    sellAmount: "uint256"
+    buyAmount: "uint256"
+    validTo: "uint32" = int(int(time.time()) + 240)
+    appData: "bytes32" = HexBytes(
+        "0x000000000000000000000000000000000000000000000000000000000000ca1f")
+    feeAmount: "uint256"
+    kind: "string"
+    partiallyFillable: "bool" = False
+    sellTokenBalance: "string" = "erc20"
+    buyTokenBalance: "string" = "erc20"
+    signingScheme: "string" = "eip712"
+
+    def sign(self):
+        sig = w3.eth.account.sign_message(
+            self.signable_message,
+            private_key=domains.private_key
         )
-        res.fee_amount = res.get_fee()
-        # TODO - could make method set_fee which updates field directly.
-        return res
+        return str(sig.signature.hex())
+
+    def set_json(self):
+        order = self.body_data['message']
+        order['signature'] = self.sign()
+        # TODO - We shouldn't have to do this! maybe a method called "garnish"
+        order['sellAmount'] = str(self.sellAmount)
+        order['buyAmount'] = str(self.buyAmount)
+        order['feeAmount'] = str(self.feeAmount)
```

Comment:
I see that in order to construct the Order object we have to pass along an int, but then when we want to set json, we have to cast all the values back to string. This seems strange and we should fix this (eventually) but I will let it pass for now...